### ISSUE TITLE
Add Media Revisions UI module to GovCMS

### ISSUE THREAD

**Issue by KaytooKreations** [govCMS/GovCMS#1004]
** What value does this module/package add to GovCMS?

** Is the module Drupal 10 compatible? Yes

** Please provide a brief outline of what this module does.
Allows a publisher to view revisions of a media item which is important to see who made changes to it in the past, and why

** Who does this module benefit:

[ ] content editors
[ ] site builders

[ ] developers
 

** If this module styles or alters HTML or JavaScript output, can the functionality be provided via the theme? What alternatives have you considered. - does not alter HTML and JS

** What is the maintenance and support status of the module. Describe the issue queue activity. - n/a

** What permissions are needed to utilise the module (and are any new permissions provided by the module)? - permissions that are generally applied to content editors

** Does the module modify the database structure and/or store additional metadata on nodes or other entities? If so, why? What are the risks for future updates? - No

** Is the module designed to capture anonymous user data? - No

** Is the output of the module typically fully cacheable? Would the inclusion of this module potentially render pages uncacheable. - no

** What is your assessment of the quality of this module, the contribution history of the module's maintainers, and the uptake of the module within the Drupal community? - no assessment

** Additional information - none

**Comment by thisisalistairsaccount** [comment]
Hi @KaytooKreations thanks for the submission. There's a few things not filled in your submission, so I will fill some of those to better round this out.

Module can be found at: https://www.drupal.org/project/media_revisions_ui

**What is the maintenance and support status of the module. Describe the issue queue activity.** - There are 7 open issues currently on the issue queue https://www.drupal.org/project/issues/media_revisions_ui?categories=All

**What permissions are needed to utilise the module (and are any new permissions provided by the module)?** - The module seeks permissions for each Media Type to:
- Delete media revisions
- Revert media revisions
- View media revisions
- Revert all media revisions

These can then all be defined to specific roles within the site to control who can do what. 

**What is your assessment of the quality of this module, the contribution history of the module's maintainers, and the uptake of the module within the Drupal community?** - The module fills a gap of something not available in Drupal Core. The module reports [3,750 sites report using this module](https://www.drupal.org/project/usage/media_revisions_ui) and has had its last stable release in September 2022.

**Comment by thisisalistairsaccount** [comment]
This seems like it would be a good inclusion for the GovCMS distribution, but it does seem as though there is an issue that conflicts with Media File Replace (used in the GovCMS distribution) mentioned at https://www.drupal.org/project/media_revisions_ui/issues/3196520#comment-14535595 which then leads down the rabbit hole in Drupal Core at https://www.drupal.org/project/drupal/issues/3271688#comment-14535585

Noting that Media File Replace is commonly used in the GovCMS distribution, this outstanding issue may cause concerns with us progressing this at the moment. 

I am still keen to see interest from the community on something like this, as this or something like it would give better visibility to site owners about when media was replaced.

You could potentially do this with media revisions and views, but the functionality offered in this module makes the experience like looking at revisions of content and would be functionally preferred.

**Comment by KaytooKreations** [comment]
Thanks for this Alistair

We'll go ahead with media revisions and views - Raminder has some ideas

cheers
Kathleen

**Comment by naveenvalecha** [comment]
This module has become part of the Drupal core. The Media Revision UI is added in the 10.2 https://www.drupal.org/project/drupal/issues/2911977 When GovCMS is going to update to 10.2?

Regards,
Naveen

**Comment by thisisalistairsaccount** [comment]
Hi @naveenvalecha thanks for the pickup, I didn't realise it was one of the media initiatives for Core - but that's a good thing it's been progressed for 10.2

We're currently dealing with a CK Editor 4 to 5 move which needs to be done by 15 March 2024. Hoping that after that we can then focus on a move to 10.2. On last review, there were some modules used in our distribution that still needed attention before we could move. Hoping that's less of an issue in a few weeks.

**Comment by thisisalistairsaccount** [comment]
Hi all,
Just to let you know we've started investigating a move to 10.2. We should have more information in the coming weeks :)

**Comment by thisisalistairsaccount** [comment]
Good news, as part of this week's release, GovCMS 3.12.0, we've now moved to Drupal Core 10.2.6.

This now includes the aforementioned Media Revisions UI in Core, so we don't have to include this as a standalone module.

We've also added some information in using Media Revisions to the GovCMS Service Desk Knowledge Base too to give information on how it works.

Noting this is now available, I'll close this issue off :)
